microscopic-monks/primal/engine/world.py

```diff
@@ -0,0 +1,229 @@
+import random
+from typing import Tuple, Set
+
+from kivy.graphics.instructions import RenderContext, InstructionGroup
+
+from primal.engine.perlin import sample
+from primal.engine.sprite import Sprite
+from primal.engine.feature import Feature
+
+
+class World:
+    RADIUS_WIDTH = 2
+    RADIUS_HEIGHT = 2
+    LOAD_RADIUS = 1
+
+    def __init__(self, pos: Tuple[float, float]):
+        self.world_group = InstructionGroup()
+        self.top_group = InstructionGroup()
+
+        chunk_pos = World.get_chunk_coords_from_pos(pos)
+
+        self.loaded_center = chunk_pos
+        self.seed = random.randint(0, 2 ** 32 - 1)
+        self.chunks = dict()
+
+        self.loaded_chunks = [
```

Comment:
nice use of a nested list-comprehension. I wonder if the use of World.RADIUS_WIDTH (and the rest of the class attributes) was intentional to be explicit or not?

Reply:
Yes, it was intentional. I was thinking about making the class more generic and passing radius width, height etc. as arguments, but I decided to not generalize it.

Reply:
Yes, it was intentional. I was thinking about making the class more generic and passing radius width, height etc. as arguments, but I decided to not generalize it.